src/NHibernate/Context/ThreadLocalSessionContext.cs

```diff
@@ -1,12 +1,11 @@
 using System;
 using System.Collections.Generic;
-
+using System.Threading;
 using NHibernate;
 using NHibernate.Engine;
 
 namespace NHibernate.Context
 {
-	//TODO: refactoring on this class. Maybe using MapBasedSessionContext.
```

Comment:
Good move to have ignored and removed this comment, since the `MapBasedSessionContext` implementation is flawed (thread safety issues, going to Jira that), and unneeded with your implementation.